552 big refactoring

eoyilmaz opened the pull request.

eoyilmaz committed:
[#552] Replaced references to the deprecated `threading.currentThread()` with `threading.current_thread()`.

eoyilmaz committed:
[#552] Addressed `S602` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `S307` linting errors on all source files.

eoyilmaz committed:
[#552] Fixed newly introduced bug in `DisplayCAL.wxProfileInfo.ProfileInfoFrame.OnMotion()`.

eoyilmaz committed:
[#552] Addressed `S112` linting errors on all source files.

eoyilmaz committed:
[#552] Refactored the `DisplayCAL.wxversion` module.

eoyilmaz committed:
[#552] Added tests for the `DisplayCAL.wxversion` module.

eoyilmaz committed:
[#552] Addressed `RET502` and `D212` linting errors in `DisplayCAL.wxversion` module.

eoyilmaz committed:
[#552] Addressed all linting errors in `DisplayCAL.wxversion` module.

eoyilmaz committed:
[#552] Fixed `DisplayCAL.wxversion` tests to be able to run concurrently.

eoyilmaz committed:
[#552] Addressed `S108` linting errors on all source files.

eoyilmaz committed:
[#552] Reverted the fix for the `S108` linting error `DisplayCAL.worker` as the pointed out `/tmp` is not a directory path.

eoyilmaz committed:
[#552] Addressed `YTT101` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `D402` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `D201` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `TC003` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `I001`, `RET501`, `PYI036`, `S606` and `F401` linting errors on all source files.

eoyilmaz committed:
[#552] Fixed a recently introduced bug in `DisplayCAL.worker.Worker.exec_cmd()`.

eoyilmaz committed:
[#552] Fixed a bugs in `DisplayCAL.worker` and `DisplayCAL.wxaddons` introduced with 28d8ba9.

eoyilmaz committed:
[#552] Addressed `RUF017` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RUF010` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RUF051` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PLR0402` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `D403` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PIE804` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `D212` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RSE102` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET507` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET508` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET507` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PIE790` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET506` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PIE808` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RUF021` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PLR1730` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `PLR5501` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `SIM114` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET502` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET503` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `D202` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `RET505` linting errors on all source files.

eoyilmaz committed:
[#552] Auto fixed some linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `SIM102` linting error on `DisplayCAL.patterngenerators`.

eoyilmaz committed:
[#552] Addressed `FA102` and `N818` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `S314` linting errors on all source files.

eoyilmaz committed:
[#552] Addressed `I001` linting errors on `DisplayCAL.icc_profile`.

eoyilmaz committed:
[#552] Addressed `S310` linting errors.

eoyilmaz committed:
[#552] Addressed `S306` linting errors on `DIsplayCAL.worker` module.

eoyilmaz committed:
[#552] Addressed `S104` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF003` linting errors.

eoyilmaz committed:
[#552] Addressed `D301` linting errors.

eoyilmaz committed:
[#552] Addressed `N811` linting errors.

eoyilmaz committed:
[#552] Renamed the variable `version` to `version_string` in `DisplayCAL.meta`.

eoyilmaz committed:
[#552] Capitalized all the constants in `DisplayCAL.meta`.

eoyilmaz committed:
[#552] Addressed `I001` and `UP032` linting errors.

eoyilmaz committed:
[#552] Addressed `ANN206` linting errors.

eoyilmaz committed:
[#552] Addressed `D417` linting errors.

eoyilmaz committed:
[#552] Addressed `S103` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF015` linting errors.

eoyilmaz committed:
[#552] Addressed `PLR1704` linting errors.

eoyilmaz committed:
[#552] Addressed `S311` linting errors.

eoyilmaz committed:
[#552] Moved the old test code from `DisplayCAL.wxUntetheredFrame` to `tests`, but it is disabled for now.

eoyilmaz committed:
[#552] Ignored `S324` linting errors.

eoyilmaz committed:
[#552] Converted all the constant values to uppercase in the following `DisplayCAL` modules:
  - `argyll`
  - `argyll_names`
  - `config`
  - `debug_helpers`
  - `defaultpaths`
  - `defaults`
  - `log`
  - `options`
  - `safe_print`

eoyilmaz committed:
[#552] Removed the `DisplayCAL.defaultpaths.XDG` class as it was depending on a hack (using `locals()`) to do an otherwise simple task and non of the methods were used anywhere in the system.

eoyilmaz committed:
[#552] Fixed a bug in `DisplayCAL.config` module, introduced recently with the capitalization of the constant values on `DisplayCAL.defaultpaths`.

eoyilmaz committed:
[#552] Fixed a bug in `DisplayCAL.config` module, introduced recently with the capitalization of the constant values on `DisplayCAL.defaultpaths` (2).

eoyilmaz committed:
[#552] Some more updates related to capitalization of constants.

eoyilmaz committed:
[#552] Addressed `N817` linting errors.

eoyilmaz committed:
[#552] Addressed `A001` linting errors.

eoyilmaz committed:
[#552] Addressed `ANN401` linting errors.

eoyilmaz committed:
[#552] Addressed `PLR1714` linting errors.

eoyilmaz committed:
[#552] Addressed `ANN205` linting errors.

eoyilmaz committed:
[#552] Addressed `PIE810` linting errors.

eoyilmaz committed:
- [#552] Addressed `N815` linting errors.
- [#552] Removed deprecated `wx` constants from `wxfixes`.
- [#552] Updated `DisplayCAL.meta.WX_MINVERSION` to `4.0.0`.

eoyilmaz committed:
- [#552] Addressed `N816` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF001` linting errors.

eoyilmaz committed:
[#552] Addressed `S101` linting errors.

eoyilmaz committed:
[#552] Renamed the following to lowercase to follow PEP8 naming conventions and to address `N999` linting errors:
  - `RealDisplaySizeMM` -> `real_display_size_mm`
  - `wxVRML2X3D` -> `wx_vrml_2_x3d`
  - `wxCCXXPlot` -> `wx_ccxx_plot`
  - `wxDisplayAdjustmentFrame` -> `wx_display_adjustment_frame`
  - `wxDisplayUniformityFrame` -> `wx_display_uniformity_frame`
  - `wxLUT3DFrame` -> `wx_lut_3d_frame`
  - `wxLUTViewer` -> `wx_lut_viewer`
  - `wxMeasureFrame` -> `wx_measure_frame`
  - `wxProfileInfo` -> `wx_profile_info`
  - `wxReportFrame` -> `wx_report_frame`
  - `wxenhancedplot` -> `wx_enhanced_plot`
  - `wxaddons` -> `wx_addons`
  - `wxfixes` -> `wx_fixes`
  - `wxScriptingClient` -> `wx_scripting_client`
  - `wxSynthICCFrame` -> `wx_synth_icc_frame`
  - `wxTestchartEditor` -> `wx_testchart_editor.py`
  - `wxUntetheredFrame` -> `wx_untethered_frame`
  - `wxversion` -> `wx_version`
  - `wxVisualWhitepointEditor` -> `wx_visual_whitepoint_editor`
  - `wxwindows` -> `wx_windows`

eoyilmaz committed:
[#552] Addressed more `N999` linting errors.

eoyilmaz committed:
[#552] Some residual errors are fixed from the last commit of renaming modules.

eoyilmaz committed:
[#552] Updated `Makefile` to better printout the command messages.

eoyilmaz committed:
[#552] Addressed `SIM105` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF012` linting errors.

eoyilmaz committed:
[#552] Addressed `FLY002` linting errors.

eoyilmaz committed:
[#552] Addressed `N801` linting errors.

eoyilmaz committed:
[#552] Addressed `LOG015` linting errors.

eoyilmaz committed:
[#552] Addressed `RET504` linting errors.

eoyilmaz committed:
[#552] Addressed `D100` linting errors.

eoyilmaz committed:
[#552] Addressed `D100` linting errors (2).

eoyilmaz committed:
[#552] Addressed `TC003` linting errors.

eoyilmaz committed:
[#552] Addressed `UP037` linting errors.

eoyilmaz committed:
[#552] Addressed `UP037` linting errors (2).

eoyilmaz committed:
[#552] Addressed `PERF401` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF005` linting errors.

eoyilmaz committed:
[#552] Addressed `A002` linting errors.

eoyilmaz committed:
[#552] Addressed `D105` linting errors.

eoyilmaz committed:
[#552] Updated modules that are requiring `typing.Self` to import it from `typing-extensions` package under Python <= 3.10.

eoyilmaz committed:
[#552] Addressed `S607` linting errors by selectively ignoring it for now.

eoyilmaz committed:
[#552] Addressed `D101` linting errors.

eoyilmaz committed:
[#552] Addressed `D103` linting errors.

eoyilmaz committed:
[#552] Fix bug in `DisplayCAL.log` introduced in the last commit.

eoyilmaz committed:
[#552] Fixed improper calls to `pytest.skip` as a decorator where as a `pytest.mark.skip` is intended.

eoyilmaz committed:
[#552] Addressed `D102` linting errors.

eoyilmaz committed:
[#552] Addressed `D205` linting errors.

eoyilmaz committed:
[#552] Addressed `RUF046` linting errors.

eoyilmaz committed:
[#552] Addressed `D415` linting errors.

eoyilmaz committed:
[#552] Updated `pyproject.toml` to disable some linting checks.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lib.agw.artmanager` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lib.agw.fmresources` module.

eoyilmaz committed:
[#552] Fixed a typo in `DisplayCAL.lib.agw.fmresources` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lib.agw.fourwaysplitter` and all the related modules.

eoyilmaz committed:
[#552] Fixed a typo in `DisplayCAL.lib.agw.gradientbutton` module.

eoyilmaz committed:
[#552] Addressed some remaining `FA102` linting errors.

eoyilmaz committed:
[#552] Added return type annotations of `Self` to class methods where the method is returning `self`.

eoyilmaz committed:
[#552] Ordered the type annotations on some functions/methods that is using `None`, from simple to complex, so that `None` is the first type listed in the type annotations to pursue the `unsupported operand type(s) for |: 'type' and 'NoneType'` error messages thrown under Python 3.9, which turned out to be caused by the external `pyglet.app.cocoa` module.

eoyilmaz committed:
[#552] Fixed bug in `DisplayCAL.wx_enhanced_plot` introduced with e93ccd6.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.x3dom` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lib.agw.labelbook` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lib.agw.pygauge` module.

XLR-Security commented:
Wow, you are soloing this like it's a boss from Dark Souls. Incredible work, thanks for keeping this project alive!

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.argyll_cgats` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.argyll_names` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.argyll_rgb2xyz` module.

eoyilmaz committed:
[#552] Updated an occasionally failing test in `test_worker` module to create a `wx.App` to see if this is going to solve the problem.

eoyilmaz committed:
[#552] Updated `DisplayCAL.argyll.set_argyll_bin()` to create a `wx.App` to see if this is going to solve the problem of occasionally failing tests.

eoyilmaz committed:
[#552] Renamed the `NORMALISATION_FACTOR` to `NORMALIZATION_FACTOR` to use US english as PEP8 requires it.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.argyll` module.

eoyilmaz committed:
[#552] Fixed `requirements.txt` to include `pyglet` which was accidentally included in `requirements-dev.txt` and that was the main issue causing the audio subsystem not working on installed version of DisplayCAL. Also referenced `pyglet<=2.1.3` for `Python 3.9` as the newer versions are not working with `Python 3.9` due to unsupported annotation related issues.

eoyilmaz committed:
[#552] Fixed `DisplayCAL.wx_windows.LogWindow.OnDestroy()` to ignore the `RuntimeError`.

eoyilmaz committed:
[#552] Fixed `DisplayCAL.wx_windows.ProgressDialog.anim_fadein()` to to call the `AnimatedBitmap.SetBitmaps()` with the `range_` argument instead of the `range` as it is recently renamed.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.audio` module.

eoyilmaz committed:
[#552] Fixed a bug in `DisplayCAL.argyll_cgats` introduced with `56f752d`.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.ccmx` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.cgats` module.

eoyilmaz committed:
[#552] Added `DisplayCAL.colorspace_to_vrml` module.

eoyilmaz committed:
[#552] Moved all colorspace to VRML related classes in `DisplayCAL.cgats` to `DisplayCAL.colorspace_to_vrml` module.

eoyilmaz committed:
- [#552] Added tests for `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML` and derivatives.
- [#552] Added the `DisplayCAL.colorspace_to_vrml.CAT` enum class for different chromatic adaptation transform values.

eoyilmaz committed:
[#552] Added more enum values to `DisplayCAL.colorspace_to_vrml.CAT` enum class and updated tests.

eoyilmaz committed:
[#552] Fixed a `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML` class test.

eoyilmaz committed:
[#552] Fixed a `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML` class test (2).

eoyilmaz committed:
[#552] Added the `DisplayCAL.colorspace_to_vrml.ColorSpace` enum class for different colorspace values.

eoyilmaz committed:
[#552] Updated `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML` to utilize the `ColorSpace` enum class.

eoyilmaz committed:
[#552] Fixed some tests related to `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML`.

eoyilmaz committed:
[#552] Updated the `test` target in `Makefile` to show the coverage report for the `DisplayCAL` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.chromecast_patterngenerator` module.

eoyilmaz committed:
[#552] Renamed `chromecast_patterngenerator` module to `chromecast_pattern_generator`.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.colord` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.colormath` module (except C901).

eoyilmaz committed:
- [#552] Reduced the complexity of  `DisplayCAL.colormath.linmin()` by dividing it into 2 separate functions, but can't reduce the complexity of those two anymore without spending too much effort.
- [#552] Reduced the complexity of `DisplayCAL.colormath.special_pow()` quite a bit to the accepted levels.

eoyilmaz committed:
[#552] Fixed the `RUF003` linting error in `DisplayCAL.colormath`.

eoyilmaz committed:
[#552] Fixed several linting errors in `colorspace_to_vrml`, `wx_lut3d_frame` and `wx_testchart_editor` modules of `DisplayCAL`.

eoyilmaz committed:
[#552] Fixed `DisplayCAL.colorspace_to_vrml.ColorSpaceToVRML.VALID_COLOR_SPACE_NAMES` attribute value to be a `tuple` instead of a generator.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.config` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.cubeiterator` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.debughelpers` module.

eoyilmaz committed:
[#552] Replaced the `NO-BREAK SPACE` characters with `SPACE` on some source files.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.defaultpaths` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.demjson_compat` module.

eoyilmaz committed:
[#552] Added tests and fixed bugs in `DisplayCAL.demjson_compat` module.

eoyilmaz committed:
[#552] In`DisplayCAL.demjson_compat` module, Implemented the `decode()` functionality as a class, which cleaned up the code immensely.

eoyilmaz committed:
[#552] Fixed most of the linting errors, except the 35 of the 51 original C901 errors, in `DisplayCAL.display_cal` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.edid` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.embeddedimage` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.encodedstdio` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.encoding` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.floatspin` module.

eoyilmaz committed:
[#552] Fixed all linting errors (except one C901 error, which wouldn't really worth wasting time to) in `DisplayCAL.freeze` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `util.scale_cal` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.x3dom` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.wx_windows` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.wx_testchart_editor` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.taskscheduler` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.madvr` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.imfile` module.

eoyilmaz committed:
[#552] Removed calls to `locals()` in `DisplayCAL.cubeiterator` module.

eoyilmaz committed:
[#552] Fixed all linting errors, except `C901` errors, in `DisplayCAL.icc_profile` module. `C901` will be handled along with the other modules.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.imfile` module.

eoyilmaz committed:
[#552] Fixed bug in `DisplayCAL.wx_testchart_editor` module introduced with 58356ef.

eoyilmaz committed:
[#552] Addressed some formatting errors or improvements introduced with recent commits.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.jsondict` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.jspacker` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.lazydict` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.localization` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.log` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.madvr` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.main` module.

eoyilmaz committed:
[#552] Fixed import statements in `DisplayCAL.wx_scripting_client` and `DisplayCAL.wx_visual_whitepoint_editor` modules.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.meta` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.multiprocess` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.network` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.patterngenerator` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.postinstall` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.profile_loader` module (except a couple of left over C901's).

eoyilmaz committed:
[#552] Addressed all linting errors in `DisplayCAL.real_display_size_mm` module.
[#552]  Renamed the following functions in `DisplayCAL.real_display_size_mm` module:
  - `GetXRandROutputXID` to `get_xrandr_output_xid`.
  - `RealDisplaySizeMM` to `real_display_size_mm`.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.report` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.safe_print` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.setup` module (except C901's).

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.systrayicon` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.taskbar` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.taskscheduler` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.util_dbus` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.util_decimal` module and added tests.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.util_dict` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.util_http` module.

jaminmc committed:
[#552] Overlay project CI workflow and ES/IT translations from ArgyllCMS-3.5.0-more.

Keep the 552 refactor baseline while preserving our tested workflow and localized strings for Spanish and Italian.

jaminmc committed:
[#552] Harden wx UI fallbacks and Python 3.14 packaging flow

Refactor crash-prone wx paths to use safer control creation and robust Argyll download/version error handling so startup and profile-info flows fail gracefully instead of trapping. Align packaging, dependencies, and install docs with Python 3.14 support and current build behavior.

jaminmc committed:
[#552] Harden Argyll setup flow and 3.5.0 docs integration

Combine Argyll setup and GUI robustness updates with the ArgyllCMS 3.5 integration/docs refresh into a single cohesive changeset for easier review and release tracking.

eoyilmaz committed:
Merge pull request #645 from jaminmc/552-refactor-with-our-l10n-ci

Integrate ArgyllCMS 3.5.0 updates, specplot option, and docs/changelog refresh

eoyilmaz committed:
[#552] Fixed all linting errors in `DisplayCAL.util_io` module.

jaminmc committed:
Integrate Windows MSCMS wrapper with safe dependency and API fixes

Port the PR #591 MSCMS manager/wrapper and tests into the #552 branch while preserving current cross-platform behavior and dependency set. Fix retry semantics, WCS API signature/size handling, and English test docstrings for safer Windows profile management integration.

Co-authored-by: Anulap <perpetuallybored@yandex.ru>

eoyilmaz committed:
[#552] Trying to fix the `typing_extensions` related ImportErrors by not requiring it in runtime.

eoyilmaz committed:
[#552] Update Python and wxPython versions in CI configuration with exclusions.

jaminmc committed:
[#552] Harden download recovery flow and localize retry/archive messages.

Improve download robustness with retry/backoff, hash verification, atomic file replacement, and cleanup of invalid archives while re-prompting users to retry. Add translation keys for retry-progress and archive-delete messages across all supported locales.

eoyilmaz committed:
[#552] Update wxPython version check to include 4.2.4 and 4.2.5 in CI configuration.

eoyilmaz committed:
[#552] Remove Python versions 3.12, 3.13, and 3.14 from CI configuration for Windows as `msilib` is deprecated and needs bigger work to support those Python versions.

eoyilmaz committed:
[#552] Refactor CI configuration

...to include specific wxPython versions for each Python version and simplify the matrix setup.

eoyilmaz committed:
[#552] Implement retry logic for fetching ArgyllCMS latest version to handle transient network issues.

eoyilmaz committed:
[#552] Refactor error message handling in tests to simplify retrieval based on Python version.

jaminmc committed:
Improve Web @ localhost stability and continuity

Fix Web @ localhost polling and color updates so browser clients recover from transient errors, preserve session continuity, and render accurate patch colors. Also align Linux and macOS install docs with the lowercase `displaycal-*.whl` wheel naming used by current build output.

jaminmc committed:
[#552] Add missing typing_extensions build dependency for isolated builds

Include typing_extensions in build-system requirements for Python < 3.11 so PEP 517 isolated builds do not fail when setup imports DisplayCAL.util_os.

eoyilmaz committed:
[#552] Updated CI workflow to set the minimum tested wxPython versions to 4.2.2 (removed 4.2.0 and 4.2.1).

eoyilmaz committed:
[#552] Updated `.gitignore` to ignore `.env` files.

eoyilmaz committed:
[#552] Fixed #641.

We still have issues especially under Windows when the display scaling is greater than 100%. Hopefully this commit fixes the last bit of errors.

eoyilmaz committed:
[#552] Add null check for `event` in `measurement_mode_ctrl_handler` to prevent errors.

eoyilmaz committed:
Merge pull request #646 from jaminmc/552-MSCMS-pr-591

[#552] Integrate MSCMS PR #591 with Windows Python 3.13+ build compatibility fixes

eoyilmaz committed:
[#552] Fixed linting errors in `mscms` module.

eoyilmaz committed:
[#552] Refactor file handling to use built-in `open()` with encoding instead of `codecs.open()`.

eoyilmaz committed:
Merge pull request #647 from jaminmc/fixwebatlocal

[#552] Improve Web @ localhost stability and continuity

eoyilmaz committed:
[#552] Added back the `PyYAML` as a dependency.

eoyilmaz committed:
[#552] Fix `mscms` module for Python 3.9.

eoyilmaz committed:
[#552] Added `CIE2015_2` CAT matrix to `DisplayCAL.colormath.CAT_MATRICES` (why was this missing???).

eoyilmaz committed:
[#552] `setup_argyll()` fixture is a session level fixture now. This should reduce the test duration by a minute or so (30% reduction).

eoyilmaz committed:
[#552] `DisplayCAL.util_os.mksfile()` now returns only the file path and does not try to create a file handle and return it at all.

eoyilmaz committed:
[#552] Fixed `DisplayCAL.worker.Worker._download()` to work like it suppose to be.

eoyilmaz committed:
[#552] Fixed some failing tests under Windows.

eoyilmaz committed:
[#552] Possible fix for CI tests running on macOS.

eoyilmaz committed:
[#552] Possible fix for CI tests running on macOS (2).

eoyilmaz committed:
[#552] Possible fix for CI tests running on macOS (3).

eoyilmaz committed:
[#552] Update CI test conditions for macOS and add skip for random failures.

eoyilmaz committed:
[#552] Fix syntax error in skip condition for macOS CI tests.

eoyilmaz committed:
[#552] Skip MeasurementFileCheckSanityDialog test on CI macOS due to failures.

eoyilmaz committed:
[#552] Updated `pytest` command to list the slowest 50 tests.

eoyilmaz committed:
[#552] Fixed `project.license` definition in `pyproject.toml`.

eoyilmaz committed:
[#552] Fixed linting errors in `mscms_types` module.

eoyilmaz committed:
[#552] Fixed all linting errors in `mscms_wrapper.py`

eoyilmaz committed:
[#552] Update code formatting on several files.

eoyilmaz committed:
[#552] Updated CI tests for macOS to include different wxPython versions.

Now that the tests are stabilized, added matrix of wxPython versions for macOS tests as in Linux and Windows tests.

eoyilmaz committed:
[#552] Fix CI macos code coverage report name.

eoyilmaz committed:
[#552] `git_version` is renamed to `git_revision_hash_short` in `setup.py`.

eoyilmaz committed:
[#552] Initial commit for nightly builds.

eoyilmaz commented:
Finally, this is ready to be merged to the develop branch. Although, there is a ton of work that needs to be done and I still need to go over a lot of things, there is no reason to hold this incredibly huge PR back anymore...

Thanks for everybody who participated 👏 ❤️
